### ISSUE TITLE
Steam high cpu load when exiting specific games

### ISSUE THREAD

**Issue by doctortee** [ValveSoftware/steam-for-linux#6811]
#### Your system information

* Steam client version (build number or date): steam package version 1576908998 (20/12/19)
* Distribution (e.g. Ubuntu): Debian testing
* Opted into Steam client beta?: [Yes/No] Yes (same issue on stable)
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Similar to issue #5493 , but as mentioned here (https://github.com/ValveSoftware/steam-for-linux/issues/5493#issuecomment-565597689) , they seem to focus on the "steam-runtime-system-info" problem, which seems not to be the issue here.

When I play to Warframe (It might be specific, but I won't say that) for some time and then close the game, a specific Steam thread is going crazy and uses 100% of one of my cores.
I always have to kill these process to quit Steam as they don't exit themselves when I close it.
In the issue mentioned earlier, they say it comes from "steam-runtime-system-info", but even if I rename the file, the high CPU load is still present.

I tried to switch to stable, same issue.
While this definitely seems to be related with #5493 , it only occurs when I exit this specific game, and the workaround mentioned aren't working.

Note that I tried other games, and I don't have that issue, whether it uses Proton or not.

#### Steps for reproducing this issue:

1. Start Steam
2. Start concerned game
3. Play for some time
4. Exit the game
5. Profit.

#### Additional logs:

Logs starting when I close the game. Ask if you need the whole log, but as I often switch between windows, it spams messages about controller.

`Exiting app 230410
No cached sticky mapping in ActivateActionSet.No cached sticky mapping in ActivateActionSet.[2020-01-05 13:28:37] Background update loop checking for update. . .
[2020-01-05 13:28:38] Recherche de nouvelles mises à jour...
[2020-01-05 13:28:38] Downloading manifest: client-download.steampowered.com/client/steam_client_publicbeta_ubuntu12
[2020-01-05 13:28:38] Download skipped by HTTP 304 Not Modified
[2020-01-05 13:28:38] Nothing to do
../common/pipes.cpp (774) : Assertion Failed: stalled cross-thread pipe
../common/pipes.cpp (774) : Assertion Failed: stalled cross-thread pipe
Installing breakpad exception handler for appid(steam)/version(1576908998)
assert_20200105144138_33.dmp[13984]: Uploading dump (out-of-process)
/tmp/dumps/assert_20200105144138_33.dmp
steamclient.cpp (793) : Assertion Failed: bufRet.TellPut() == sizeof(uint8)
steamclient.cpp (793) : Assertion Failed: bufRet.TellPut() == sizeof(uint8)
assert_20200105144138_33.dmp[13984]: Finished uploading minidump (out-of-process): success = yes
assert_20200105144138_33.dmp[13984]: response: CrashID=bp-51cbe4ae-b5d9-4cef-b9c2-ef2452200105
assert_20200105144138_33.dmp[13984]: file ''/tmp/dumps/assert_20200105144138_33.dmp'', upload yes: ''CrashID=bp-51cbe4ae-b5d9-4cef-b9c2-ef2452200105''
Thread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut downThread "CJobMgr::m_WorkThreadPool:0" (ID 8365) failed to shut down`

**Comment by kisak-valve** [comment]
Hello @doctortee, can you check if this is Steam's shader precaching doing work by going to `Steam` -> `Settings` -> `Shader Pre-Caching` -> uncheck `Enable Shader Pre-Caching`, then test if there is an effect. If it does, then this is the expected behavior, but the worker thread probably shouldn't stick around if the steam client is closed.

**Comment by doctortee** [comment]
The issue seems to disappear when I disable Shader Pre-Caching. (Tested once).

**Comment by wylel** [comment]
I too was having this issue. Same exact problem as doctortee but on different distro and hardware. 

This issue only happened to Warframe with Pre-Cached Shaders. Turning that off did the trick.

With that being said, is there a way, or can there be a way, to turn Pre-Cached Shaders off on a game-by-game basis?

**Comment by danilo-bc** [comment]
I had the same issue with Path of Exile, and not with other games. Disabling Shader Pre-Caching solved this issue.

**Comment by userofryzen** [comment]
I am having this problem with rise of the tomb raider too.. I haven't checked if disabling tthis option is the fix of this

**Comment by userofryzen** [comment]
anybody has this error too?

**Comment by son-link** [comment]
> I am having this problem with rise of the tomb raider too.. I haven't checked if disabling tthis option is the fix of this

Yes, i have the same issue with the lastets client and AMDGPU drivers

**Comment by Saroumane** [comment]
I just had this issue (it's random for me) when trying to exit Steam client. (And I have already disabled "Shader Pre-Caching") :

`.steam/error.log` is continuously filled with `Thread "CJobMgr::m_WorkThreadPool:0" (ID 2226155) failed to shut down `  for 30min now
`.steam/logs/shader_log.txt ` had no new data for 30min, ending with "`AppID 253230 exited. / Finding Mesa v2 cache files, base directory`"
I had to kill Steam process.

system information
- Steam client version : 20 Dec 2020
- Distribution (e.g. Ubuntu): Ubuntu 20.10
- Opted into Steam client beta?: No
- Have you checked for system updates?: Yes
- AMD RX 5700 XT (driver amdgpu)
- Mesa 20.3.2 - kisak-mesa PPA

**Comment by kndgs** [comment]
Having this issue, fresh steam install including user folder, pre-caching is disabled. Arch, nvidia, all up to date.
Don't even need to launch any games for this to happen.

**Comment by KenTheWhaleGoddess** [comment]
Got this as well, when **disabling** shader pre-caching, then clearing Download cache.

**Comment by KenTheWhaleGoddess** [comment]
Chat GPT since no recent comments...

```The message you're seeing, "Thread 'CJobMgr::m_WorkThreadPool:0' (ID 4902) failed to shut down," is likely coming from a software or application that uses multithreading. This message indicates that a thread with the given name and ID failed to shut down properly during the program's termination or under specific circumstances.```
